File: packages/modules/api/lichess.org/src/lichess.dto.ts
export enum Variant {
  STANDARD = 'standard',
  CHESS_960 = 'chess960',
  CRAZY_HOUSE = 'crazyhouse',
  ANTI_CHESS = 'antichess',
  ATOMIC = 'atomic',
  HORDE = 'horde',
  KING_OF_THE_HILL = 'kingOfTheHill',
  RACING_KINGS = 'racingKings',
  THREE_CHECK = 'threeCheck',
  FROM_POSITION = 'fromPosition',
}

export type CloudEvaluation = {
  error?: string;
  depth: number;
  fen: string;
  knodes: number;
  pvs: { mate?: number; cp?: number; moves: string }[];
};

File: packages/modules/api/python/src/api/lichess_dto.py
from dataclasses import dataclass
from enum import StrEnum


class Variant(StrEnum):
    STANDARD = "standard"
    CHESS_960 = "chess960"
    CRAZY_HOUSE = "crazyhouse"
    ANTI_CHESS = "antichess"
    ATOMIC = "atomic"
    HORDE = "horde"
    KING_OF_THE_HILL = "kingOfTheHill"
    RACING_KINGS = "racingKings"
    THREE_CHECK = "threeCheck"
    FROM_POSITION = "fromPosition"


@dataclass
class PV:
    cp: int | None = None
    mate: int | None = None
    moves: str | None = None


@dataclass
class CloudEvaluation:
    depth: int
    fen: str
    knodes: int
    pvs: list[PV]
    error: str | None = None
